### PR TITLE
Fix bug in enum handling, plus test

### PR DESCRIPTION
**Pull request by chrisdunelm**

### CODE REVIEW

File: Google.Api.Generator.Tests/ProtoTests/MethodSignatures/MethodSignatures.proto [hunk 1]
```diff
@@ -89,6 +89,11 @@ message MessageRequest {
   repeated Msg repeated_required = 4 [(google.api.field_behavior) = REQUIRED];
 }
 
+enum TopLevelEnum {
+	DEFAULT = 0;
```
Comment (jskeet) [inline Google.Api.Generator.Tests/ProtoTests/MethodSignatures/MethodSignatures.proto]: That's a lot of whitespace. Outdent a bit?

File: Google.Api.Generator/Utils/Typ.cs [hunk 2]
```diff
@@ -59,7 +59,7 @@ private sealed class Array : Typ
 
         private sealed class FromManual : Typ
         {
-            public FromManual(string ns, string name, bool isEnum = false) => (Namespace, Name, IsEnum) = (ns, name, IsEnum);
+            public FromManual(string ns, string name, bool isEnum = false) => (Namespace, Name, IsEnum) = (ns, name, isEnum);
```
Comment (jskeet) [inline Google.Api.Generator/Utils/Typ.cs]: Ouch. It really feels like the compiler should have warned you about that.

File: Google.Api.Generator.Tests/ProtoTests/MethodSignatures/MethodSignatures.proto (hunk 1 above)
Comment (chrisdunelm) [reply Google.Api.Generator.Tests/ProtoTests/MethodSignatures/MethodSignatures.proto]: Tabs :( Done

File: Google.Api.Generator/Utils/Typ.cs (hunk 2 above)
Comment (chrisdunelm) [reply Google.Api.Generator/Utils/Typ.cs]: Yes, I'm surprised it didn't warn.

**Pull request merged**